src/tools/gophish_export.py

```diff
@@ -253,6 +253,86 @@ def get_application(rawEvent):
     return application
 
 
+def find_unique_target_clicks_count(clicks):
+    """Return the number of unique clicks in a click dict."""
+    uniq_list = list()
+    for click in clicks:
+        if click["user"] not in uniq_list:
+            uniq_list.append(click["user"])
+    return len(uniq_list)
+
+
+def write_campaign_summary(api, assessment_id):
+    """Output a campaign summary report to JSON, console, and a text file."""
+    campaign_ids = get_campaign_ids(api, assessment_id)
+    campaign_data_template = "campaign_data.json"
+    campaign_summary_json = assessment_id + "_campaign_data.json"
+    campaign_summary_textfile = (
+        assessment_id + "_summary_" + str(datetime.now()) + ".txt"
+    )
```

Comment:
f-strings are preferable just about everywhere (except in logging statements).  
Also, do you really want that space (from `str(datetime.now())` in your filenames?  I'd use the same format as you use later on:
```suggestion
    campaign_summary_json = f"{assessment_id}_campaign_data.json"
    campaign_summary_textfile = f"{assessment_id}_summary_{datetime.strftime(datetime.now(), '%Y-%m-%dT%H:%M:%S')}.txt"
```

Reply:
The reason behind the use of datetime was by the desire to keep a separate file for each time the export is run. They did not want repeated runs of the export script to result in the outputted text file being overwritten.

Reply:
> The reason behind the use of datetime was by the desire to keep a separate file for each time the export is run. They did not want repeated runs of the export script to result in the outputted text file being overwritten.

I don't have any issue with putting a timestamp in the filename- I just try to avoid creating filenames with spaces in them.

Reply:
See 5816a24f37e9a600f2b55946de18b7acb8d0da07